test/sdk/fixed_bytes_contract_test.go

```diff
@@ -0,0 +1,78 @@
+package sdk
+
+import (
+	"context"
+	sdkContext "github.com/orbs-network/orbs-contract-sdk/go/context"
+	"github.com/orbs-network/orbs-network-go/crypto/hash"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-network-go/test/sdk/contracts/fixed_bytes"
+	"github.com/orbs-network/orbs-network-go/test/with"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+const ContractName = "TestBytesContract"
+const ContractFakeCode = " "
+
+func mockForTest() *sdkContext.ContractInfo {
+	return &sdkContext.ContractInfo{
+		PublicMethods: fixed_bytes.PUBLIC,
+		SystemMethods: fixed_bytes.SYSTEM,
+		Permission:    sdkContext.PERMISSION_SCOPE_SERVICE,
+	}
+}
+
+func TestVm_CanCompileContractWithFixedBytes(t *testing.T) {
+	bytes20 := [20]byte{0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01,
+		0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01}
+	bytes32 := [32]byte{0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x04,
+		0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x01, 0x02, 0x03, 0x04}
+	almostEmptyBytes20 := [20]byte{0x01}
+
+	test.WithContext(func(ctx context.Context) {
+		with.Logging(t, func(parent *with.LoggingHarness) {
+
+			harness := newVmHarness(parent.Logger)
+			harness.compiler.ProvideFakeContract(mockForTest(), ContractFakeCode)
+
+			txs := []*protocol.SignedTransaction{
+				//					builders.Transaction().WithMethod(WorkingContractName, WorkingMethodName).WithArgs(hash.Make32BytesWithFirstByte(6)).Build(),
+				builders.Transaction().WithMethod("_Deployments", "deployService").
+					WithArgs(ContractName, uint32(protocol.PROCESSOR_TYPE_NATIVE), []byte(ContractFakeCode)).
+				    Build(),
+				builders.Transaction().WithMethod(ContractName, "setAddress").WithArgs(bytes20).Build(),
+				builders.Transaction().WithMethod(ContractName, "setHash").WithArgs(bytes32).Build(),
+				builders.Transaction().WithMethod(ContractName, "getAddress").WithArgs().Build(),
+				builders.Transaction().WithMethod(ContractName, "getHash").WithArgs().Build(),
+				builders.Transaction().WithMethod(ContractName, "setAddress").WithArgs(almostEmptyBytes20).Build(),
+				builders.Transaction().WithMethod(ContractName, "getAddress").WithArgs().Build(),
+			}
+
+			out, err := harness.vm.ProcessTransactionSet(ctx, &services.ProcessTransactionSetInput{
```

Comment:
as discussed, please hide all irrelevant stuff inside the harness, and create a function `h.process(ctx, tx1, tx2, tx3)`